### PR TITLE
New external ucs related params in ptu preloaded pt snapshot pt and external ucs ON OFF and ucs informing HMI

### PR DESCRIPTION
**Pull request by okozlovlux**

### CODE REVIEW

File: src/components/policy/policy_external/src/update_status_manager.cc [hunk 1]
```diff
@@ -61,17 +61,17 @@ UpdateStatusManager::~UpdateStatusManager() {
 
 void UpdateStatusManager::ProcessEvent(UpdateEvent event) {
   sync_primitives::AutoLock lock(status_lock_);
-  LOG4CXX_DEBUG(logger_,
+  /*LOG4CXX_DEBUG(logger_,
```
Comment (LuxoftAKutsan) [inline src/components/policy/policy_external/src/update_status_manager.cc]: @okozlovlux avoid commented code

**Comment by okozlovlux** [comment]
Rebuild required

File: src/components/application_manager/include/application_manager/message_helper.h [hunk 2]
```diff
@@ -359,15 +361,26 @@ class MessageHelper {
...
+ * @brief Send GetListOfPermissions response to HMI
+ * @param permissions Array of groups permissions
+ * @param external_consent_status External user consent status
+ * @param correlation_id Correlation id of request
+ */
+#ifdef EXTERNAL_PROPRIETARY_MODE
   static void SendGetListOfPermissionsResponse(
       const std::vector<policy::FunctionalGroupPermission>& permissions,
-      uint32_t correlation_id,
+#if EXTERNAL_PROPRIETARY_MODE
```
Comment (okozlovlux) [inline src/components/application_manager/include/application_manager/message_helper.h]: @jacobkeeler fixed in 718663d

File: src/components/interfaces/HMI_API.xml [hunk 3]
```diff
@@ -1235,17 +1239,19 @@
...
-  </param>
+	<param name="name" type="String" mandatory="true">
+	  <description>Code of message of user-friendly text about functional group to be allowed/disallowed</description>
+	</param>
+	<param name="id" type="Integer" mandatory="true">
+	  <description>Id of corresponding functional group, needed by SDL.</description>
+	</param>
+	<param name="allowed" type="Boolean" mandatory="false">
+	  <description>Specifies whether functionality was allowed/disallowed. If ommited - no information about User Consent is yet found for app.</description>
+	</param>
```
Comment (okozlovlux) [inline src/components/interfaces/HMI_API.xml]: @jacobkeeler fixed in 718663d

File: src/components/policy/policy_regular/src/policy_table/types.cc [hunk 4]
```diff
@@ -172,7 +172,7 @@ Json::Value ApplicationParams::ToJsonValue() const {
   Json::Value result__(PolicyBase::ToJsonValue());
   impl::WriteJsonField("groups", groups, &result__);
   impl::WriteJsonField("nicknames", nicknames, &result__);
-  impl::WriteJsonField("AppHMIType", AppHMIType, &result__);
+  impl::WriteJsonField("AppHmiType", AppHMIType, &result__);
```
Comment (okozlovlux) [inline src/components/policy/policy_regular/src/policy_table/types.cc]: @jacobkeeler fixed in 718663d

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 5]
```diff
@@ -614,61 +685,98 @@ void PolicyHandler::OnGetUserFriendlyMessage(
...
+  const bool is_connection_key_valid = is_app_registered && connection_key;
+
+  const std::vector<policy::FunctionalGroupPermission> permissions =
+      is_connection_key_valid ? CollectAppPermissions(connection_key)
+                              : CollectRegisteredAppsPermissions();
+
+  if (permissions.empty() && is_connection_key_valid) {
+    LOG4CXX_ERROR(logger_,
+                  "No permissions found for application with connection key:"
+                      << connection_key);
```
Comment (jacobkeeler) [inline src/components/application_manager/src/policies/policy_handler.cc]: Why is an error response not sent in this case?

**Comment by jacobkeeler** [comment]
My Core log and policy table:
[SmartDeviceLinkCore.txt](https://github.com/smartdevicelink/sdl_core/files/1023813/SmartDeviceLinkCore.txt)
[sdl_preloaded_pt.json](https://github.com/smartdevicelink/sdl_core/files/1023811/sdl_preloaded_pt.txt)

**Comment by LuxoftAKutsan** [comment]
@jacobkeeler please use hmi from https://github.com/smartdevicelink/sdl_hmi/pull/12

**Pull request merged**